src/main/scala/trace/TraceEncoder.scala

```diff
@@ -0,0 +1,423 @@
+// See LICENSE.Berkeley for license details.
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.trace
+
+import chisel3._
+import chisel3.util._
+import scala.math.min
+
+import org.chipsalliance.cde.config.Parameters
+
+case class TraceEncoderParams(
+  coreParams: TraceCoreParams,
+  bufferDepth: Int,
+  encoderBaseAddr: BigInt,
+  useArbiterMonitor: Boolean,
+  // a seq of functions that takes a parameter and returns a lazymodule and a target id
```

Comment:
Can you explain what the target id is for in a comment? I thnk that is not obvious